src/command_modules/azure-cli-datalake/azure/cli/command_modules/datalake/store/custom.py

```diff
@@ -0,0 +1,257 @@
+# account customiaztions
+def list_adls_account(client, resource_group_name=None):
+    account_list = client.list_by_resource_group(resource_group_name=resource_group_name) \
+        if resource_group_name else client.list()
+    return list(account_list)
+
+def create_adls_account(client, 
+                        resource_group_name,
+                        account_name,
+                        location,
+                        default_group = None,
+                        tags = None,
+                        encryption_type = None,
+                        key_vault_id = None,
+                        key_name = None,
+                        key_version = None,
+                        disable_encryption = False,
+                        tier = None):
+    create_params = DataLakeStoreAccount(location)
+    if tags:
+        create_params.tags = tags
+
+    if default_group:
+        create_params.default_group = default_group
+
+    if tier:
+        create_params.new_tier = tier
+    if not disable_encryption:
+        identity = EncryptionIdentity()
+        config = EncryptionConfig()
+        if not encryption_type:
+            logger.warning('No Encryption type passed in, defaulting to Service managed encryption. To opt out, explicitly pass in --disable_encryption')
+            encryption_type = EncryptionConfigType.service_managed
+        if encryption_type == EncryptionConfigType.user_managed:
+            if not key_name or not key_vault_id or not key_version:
+                raise CLIError('For user managed encryption, --key_vault_id, --key_name and --key_version are required parameters and must be supplied.')
+            config.key_vault_meta_info = KeyVaultMetaInfo(key_vault_id, key_name, key_version)
+        else:
+            if key_name or key_vault_id or key_version:
+                logger.warning('User supplied Key Vault information. For service managed encryption user supplied Key Vault information is ignored.')
+        
+        config.type = encryption_type
+        create_params.encryption_config = config
+        create_params.identity = identity
+    else:
+        create_params.encryption_state = EncryptionState.disabled
+        create_params.identity = None
+        create_params.encryption_config = None
+
+    return client.create(resource_group_name, account_name, create_params)
+
+def update_adls_account(client,
+                        account_name,
+                        tags = None,
+                        resource_group_name = None,
+                        default_group = None,
+                        firewall_state = None,
+                        allow_azure_ips = None,
+                        trusted_id_provider_state = None,
+                        tier = None):
+    if not resource_group_name:
+        resource_group_name = _get_resource_group_by_account_name(client, account_name)
+
+    update_params = DataLakeStoreAccountUpdateParameters()
```

Comment:
I believe our service is not quite using a true patch, and it also does some strange validation if you try to change things. We can look into changing this once I get tests up and running for these scenarios to make sure that they behave as expected.